crates/sidevm/host-runtime/src/env/wasi_env.rs

```diff
@@ -34,6 +36,43 @@ macro_rules! wasi_try {
     }};
 }
 
+/// Like the `try!` macro or `?` syntax: returns the value if the computation
+/// succeeded or returns the error value. Results are wrapped in an Ok
+macro_rules! wasi_try_ok {
+    ($expr:expr) => {{
+        let res: Result<_, Errno> = $expr;
+        match res {
+            Ok(val) => val,
+            Err(err) => {
+                return Ok(err);
+            }
+        }
+    }};
+}
+
+/// Like `wasi_try` but converts a `MemoryAccessError` to a `wasi::Errno`.
+macro_rules! wasi_try_mem_ok {
+    ($expr:expr) => {{
+        wasi_try_ok!($expr.map_err(mem_error_to_wasi))
+    }};
+}
+
+/// Like `wasi_try` but converts a `MemoryAccessError` to a `wasi::Errno`.
+macro_rules! wasi_try_block_ok {
```

Comment:
Isn't it the same as `wasi_try_ok!(Ok($expr))`?

Reply:
No, this macro allows  the inner block to use `?` syntax on a `Result<_, Errno>`.
Updated the doc comment.